### PR TITLE
Added Jacoco to modules

### PR DESCRIPTION
**Pull request by mattognibene**
Added test coverage through Jacoco. Supports multiple flavors.
Run the `test<AppVariant>UnitTestCoverage` task to generate a report. The report will be found in the associated module's /build/reports/jacoco directory.

### CODE REVIEW

File: {{ cookiecutter.repo_name }}/gradle/coverage.gradle [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+                    task "${testTaskName}CoverageVerification"(type: JacocoCoverageVerification, dependsOn: "jacoco${testTaskName}Coverage") {
+                        sourceDirectories = files([mainSrc])
+                        classDirectories = files([debugTree])
+                        executionData = files("${buildDir}/jacoco/${testTaskName}.exec")
+                        violationRules {
+                            failOnViolation = true
+                            rule {
+                                limit {
+                                    minimum = 1
```
Comment (tetedoie) [inline {{ cookiecutter.repo_name }}/gradle/coverage.gradle]: This mean, it will require 100% coverage (and fail if not matched), we should document that.

**Review by tetedoie — COMMENTED** [review]
Looks good to me, it's addressing -> https://github.com/Raizlabs/android-template/issues/58.

File: {{ cookiecutter.repo_name }}/gradle/coverage.gradle (hunk 1 above)
Comment (jonduran3000) [reply {{ cookiecutter.repo_name }}/gradle/coverage.gradle]: 100% code coverage feels a bit unrealistic, but I'll let other members of the team weigh in.
cc. @Drufyre @octohub

Comment (mattognibene) [reply {{ cookiecutter.repo_name }}/gradle/coverage.gradle]: I addressed issues with both documentation and minimum code coverage in my most recent commit.

**Review by jonduran3000 — APPROVED** [review]
Looks good to me

**Pull request merged**